src/commands/serde_utils.rs

```diff
@@ -0,0 +1,60 @@
+use serde::{Deserialize, Deserializer};
+
+pub fn deserialize_option_string_field<'de, D>(
+    field_name: &'static str, 
+    error_on_empty: bool,
+    deserializer: D,
+) -> Result<Option<String>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let opt: Option<String> = Option::deserialize(deserializer)?;
+
+    match &opt {
+        None => Ok(None),
+        Some(ref s) if s.trim().is_empty() => {
+            if error_on_empty {
+                Err(serde::de::Error::custom(format!("Field \"{}\" cannot be empty", field_name)))
+            } else {
+                Ok(None)
+            }
+        }
+        _ => Ok(opt),
+    }
+}
+
+pub fn serialize_non_empty_string_field<S>(
+    field_name: &'static str,
+    value: &String,
```

Comment:
please fix clippy warnings first...

Reply:
I wasn't aware of `cargo clippy` I'll use it from now on, thanks :smiley: